tornettools/util.py

```diff
@@ -59,3 +59,30 @@ def find_matching_files_in_dir(search_dir, filename):
                 found.append(p)
     logging.info(f"Found {len(found)} total files")
     return found
+
+# Useful for spelling integer constants multiple ways.  e.g. it can be useful
+# to spell 1 Mebi as 1048576 if that's how it's spelled out in other
+# documentation, *and* as 2 ** 20 to be able to easily verify that it's really
+# exactly 1 Mebi and not something slightly different.
+# e.g.:
+#     start_bytes = aka_int(2**20, 1048576)
+def aka_int(x, y):
+    assert(x == y)
+    return x
+
+# Written as a workaround that tgen streams don't record
+# stream['elapsed_seconds']['payload_bytes_recv'][4 * 2**20], but *do* record
+# stream['elapsed_seconds']['payload_progress_recv']['0.8'] for 5 MiB streams,
+# which is equivalent.
+#
+# More generally, this fills in any "missing" data in
+# stream['elapsed_seconds']['payload_bytes_recv'] that's present in
+# stream['elapsed_seconds']['payload_progress_recv'].
+def tgen_stream_fill_bytes_from_progress(stream):
+    if 'elapsed_seconds' not in stream:
+        return
+    for (progress_frac, progress_secs) in stream['elapsed_seconds']['payload_progress_recv'].items():
+        bytes_recv = round(float(progress_frac) * int(stream['stream_info']['recvsize']))
+        if str(bytes_recv) not in stream['elapsed_seconds']['payload_bytes_recv']:
+            # Missing from payload_bytes_recv; add it
+            stream['elapsed_seconds']['payload_bytes_recv'][str(bytes_recv)] = progress_secs
```

Comment:
I found myself wondering if this would be better handled at the source, in the tool that actually parses the tgen log files. (https://github.com/shadow/tgen/blob/main/tools/tgentools/analysis.py)

But I think it makes more sense in tornettools, since it's specific to a tor metric that we are trying to compare against.

Reply:
Hmm, yeah 4 MiB seems like a reasonable data point to add there, too, vs the current jump from 2 MiB to 5 MiB. Either way it'd be nice to handle it here for reprocessing archived output of the tgen parser though.